core/interfaces/i_registrable_field.ts

```diff
@@ -14,8 +14,11 @@ goog.declareModuleId('Blockly.IRegistrableField');
 
 import type {Field} from '../field.js';
 
-
-type fromJson = (p1: object) => Field;
+/**
+ * This should be satisfied by any function that returns a Field of some type,
+ * so `any` is acceptable.
+ */
+type fromJson = (options: {}) => Field<any>;
```

Comment:
I think this whole file can actually just be deleted (since it's type info anyway, it shouldn't be binary-breaking, but @maribethb to confirm).

This is leftover from the typescript conversion, and we don't actually use the interface originally. Originally we had something in closure that would check if a /static/ `fromJson` method exists. But there's no equivalent way to check for static methods in typescript - so it just converted it to an erroneous interface.

Reply:
Done!

This is currently being used in `core/field_registry.ts` when registering a field class, which each field class does. The field to register is then being passed to `core/registry.ts` which looks like it *would* use the info but currently abandons it via `AnyDuringMigration`. Some notion of `fromJson` could later just be added to the core `Field` class, though.